src/compiler/tsbuildPublic.ts

```diff
@@ -2244,7 +2246,7 @@ function cleanWorker<T extends BuilderProgram>(state: SolutionBuilderState<T>, p
         const inputFileNames = new Set(parsed.fileNames.map(f => toPath(state, f)));
         for (const output of outputs) {
             // If output name is same as input file name, do not delete and ignore the error
-            if (inputFileNames.has(toPath(state, output))) continue;
+            if (inputFileNames.has(toPath(state, output)) && !fileExtensionIsOneOf(output, [Extension.Dts, Extension.Dmts, Extension.Dcts])) continue;

```

Comment:
This seems like incorrect fix. This would delete d.ts files even if they are matched as input file by incorrect configuration. 

I looked at the issue and it looks like the configuration is incorrect. 

```
> tsc --explainFiles
src/file/file.tsx
  Matched by include pattern 'src/**/*.tsx' in 'tsconfig.json'
src/file/file.d.ts
  Matched by include pattern 'src/**/*.ts' in 'tsconfig.json'
```

Changing config fixes the error:
```json
{
  "compilerOptions": {
    "declaration": true
  },
  "include": ["src/**/*.tsx", "src/**/*.ts"],
  "exclude": ["src/**/*.d.ts"]
}
```